test(nifs): test for protogalaxy

cyphersnake opened the pull request:
**Motivation**
Part of #266

**Overview**
A simple test for protogalaxy folding scheme

cyphersnake commented:
Please review mock-design
I'll add other circuits tests only after approve

chaosma reviewed (commented):
There are overlap between this PR and PR 313 that caused conflicts. I also saw the vanilla unit test again here, what is the difference between this and PR 313.

cyphersnake commented:
> There are overlap between this PR and PR 313 that caused conflicts. I also saw the vanilla unit test again here, what is the difference between this and PR 313.

That's because squash policy merge PR, I did rebase by hand & push, look again, now only new code is visible here

chaosma commented on src/nifs/protogalaxy/mod.rs:
```diff
@@ -24,11 +24,11 @@ pub use accumulator::{Accumulator, AccumulatorArgs};
 /// [protogalaxy](https://eprint.iacr.org/2023/1106)
 #[derive(Clone, Debug)]
-pub struct ProtoGalaxy<C: CurveAffine> {
+pub struct ProtoGalaxy<C: CurveAffine, const L: usize> {
```
"/// L: number of instances to be folded"

chaosma commented on src/nifs/protogalaxy/tests.rs:
```diff
@@ -0,0 +1,134 @@
+            .try_collect_plonk_structure()
+            .expect("failed to collect plonk structure");
+
+        let circuit_meta = circuits_runners.map(|cr| {
```
circuit_witness is better, meta might related to circuit meta info which is confused with circuit structure

cyphersnake replied:
Will add `CircuitCtx` struct

chaosma reviewed (commented):
I added some minor stuff to be addressed.  In general, the mock design looks good. But I saw that the mock design is not finished yet.

cyphersnake commented:
CI started crashing because of an update to Rust Toolchain (our actions always uses the latest version), so I updated the toolchain in the project and fixed clippy
